javatests/arcs/android/e2e/testapp/PersonHostService.kt

```diff
@@ -58,8 +58,8 @@ class PersonHostService : ArcHostService() {
   ) : AndroidHost(
     context = context,
     lifecycle = lifecycle,
-    coroutineContext = Dispatchers.Default,
-    arcSerializationContext = Dispatchers.Default,
+    auxiliaryScope = CoroutineScope(Dispatchers.Default),
+    arcSerializationScope = CoroutineScope(Dispatchers.Default),
```

Comment:
Even in tests, let's use best practices. Here I think we want one coroutine scope parameter for the service, and we'll pass that scope in to this archost component.

Reply:
The doc/comments in this class suggests that the auxiliaryScope is not designed to be the parent scope, i.e. everything runs on it. Instead, it's really an auxiliary component which ArcHosts runs lifecycle events on it. So I think it's better explicitly call it auxiliaryScope. And in this sense, having a different scope for another kind of job (serialization) seems OK to me.

It might be better refactor this class so we move all works to auxiliaryScope and close the protected methods so subclass can't override them to move work to a different scope, which really makes the auxiliaryScope the one scope. And move serialization into its own component instead of accepting a dispatcher or scope.

I also want to know Ray's thinkings on this.

Reply:
This structure isn't really compatible with the spirit of structured concurrency. 

For a tree of components, there should be 1 top-level scope. All scopes should be descendants of that scope. Passing two scopes to a component doesn't make any sense here. 

If the serialization were a separate component, we could initialize it with the same parent scope as the component creating the Arc host, and then pass it in. However, since the functionality is baked into the Arc Host, we should use the same scope.

Reply:
The docs/comments for the class were written about a coroutineContext, not a coroutineScope. The things to consider when dealing with a coroutineScope are different than those you must considering with a coroutineContext, so the comments don't necessarily apply anymore.

Reply:
Another way to think about it: the way scopes propagate in this case is by the owning component passing its scope to the child component. Where does the second scope come from? We should not be creating bare CoroutineScopes. The scope should be a single scope created and owned by the parent component, and provided to the child component.

Reply:
The original reason for introducing the second coroutineContext parameter was that we wanted the serialization jobs to run on a different dispatcher. Since the parameter type is CoroutineContext, and CoroutineContext is a fairly overloaded concept, this detail may have been lost.

Dispatcher and scope are two different concepts. Jobs can be part of the same scope and run on different dispatchers. Jobs can run on the same dispatcher and be part of different scopes.

 In this particular case, I don't seen any need (or even a meaningful way) to introduce a second scope. The AbstractArcHost instance will be owned by some component, like an Android Service. That component will provide a single scope to the arc host.

Reply:
I understand these concepts and it'll be nice if 1 scope can do all these job. The fact is, if you look at this class, it actually creates 2 separate scopes. We need to understand our whole design works if we unify these 2 into 1 scope. The real question is: are ArcHosts' normal work and serialization the same kind of job? Because ArcHost can work without serialization and serialization has caused troubles.

What would happen if we combine them into one is unknown (i.e. if you cancel this 1 scope, you also cancel the serialization which might not finish and this could cause troubles). So I want Ray's inputs on this as he designs this part. We need to consider if 1 scope is enough, or deeper, who should decide when to do/cancel the serialization. I can totally see 2 alternative designs:
1, ArcHost cancels serialization itself so design-wise we could assume serialization. An Arc is in error or undefined state if serialization fails. In this case, we need ArcHost to own the serialization e2e and 1 scope could work.
2, The serialization is designed to be best-effort. The caller of the ArcHost provides ways (scopes, actual impls, etc) to do the serialization and ArcHost shouldn't be in a weird state if serialization fails.

Reply:
Ok, that sounds reasonable.